index.js

```diff
@@ -1,52 +1,147 @@
 const fs = require('fs');
 const url = require('url');
 
+const DEFAULT_OPTIONS = {
+    mimeType: 'application/json',
+};
+
+/**
+ * isDir is a function which take a path and return true/false
+ * @param {string} path
+ * @returns {Boolean}
+ */
+function isDir(path) {
+    return fs.lstatSync(path).isDirectory();
+}
+
+/**
+ * mergeHAR can take as many HAR as wanted and merge the log.entries
+ * @params {Object} har
+ */
+function mergeHAR(...args) {
+    const har = {
+        log: {
+            entries: [],
+        }
+    };
+    return args.reduce((acc, value) => {
+        acc.log.entries = acc.log.entries.concat(value.log.entries);
+        return acc;
+    }, har);
+}
+
+/**
+ * parse take the path to looking for HAR files and return HAR structure
+ * @param {string} path
+ */
 function parse(path) {
-    const HAR = JSON.parse(fs.readFileSync(path));
+    let HAR = {
+        log: {
+            entries: [],
+        },
+    };
+    try {
+        if (isDir(path)) {
+            fs.readdirSync(path).sort().reduce((acc, folderContent) => {
+                if (isDir(folderContent)) {
+                    acc = mergeHAR(acc, parse(folderContent));
+                }
+                if (folder.endsWith('.har')) {
+                    try {
+                        acc = mergeHAR(acc, JSON.parse(fs.readFileSync(folderContent)));
+                    } catch(e) {
+                        console.error(`can't parse the file ${folderContent}`, e);
+                    }
+                }
+                return acc;
+            }, HAR);
+        } else {
+            HAR = JSON.parse(fs.readFileSync(path));
+        }
+    } catch(e) {
+        console.error(e);
+    }
     HAR.log.entries.forEach(e => {
         e.request.url = url.parse(e.request.url);
     });
     return HAR;
 }
 
-function findInRequest(HAR, req) {
-    // console.log('find: ', req.path, req.method);
-    return HAR.log.entries.find(e => {
+/**
+ * filter request on method and path.
+ * Note if it find more than one results it will try to be more restrictive
+ * using queryString and body payload.
+ */
+function filter(HAR, req, options=DEFAULT_OPTIONS) {
+    // lets do a first pass on it.
+    const entries = HAR.log.entries.filter(e => {
         const u = e.request.url;
         if (u.pathname !== req.path) {
```

Comment:
I believe `e.request.url.pathname` will always be undefined.

Reply:
so req.path because we are on expressjs request API
and entries are parsed using nodejs url.parse API so this is pathname.

We could update for sure